src/components/TooltipNew/GuideTooltip.js

```diff
@@ -0,0 +1,272 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './style.scss';
+import PopperTooltip from './PopperTooltip';
+import TooltipPortal from './TooltipPortal';
+
+const baseClass = 'guide-tooltip';
+
+const cx = classNames.bind(styles);
+
+const addPadding = (rect, padding) => {
+  const x = Math.round(rect.x) - padding;
+  const y = Math.round(rect.y) - padding;
+  const width = Math.round(rect.width) + 2 * padding;
+  const height = Math.round(rect.height) + 2 * padding;
+  const top = y;
+  const left = x;
+  const bottom = top + height;
+  const right = left + width;
+
+  return {
+    x,
+    y,
+    width,
+    height,
+    top,
+    left,
+    bottom,
+    right
+  };
+};
+
+class VirtualReference {
+  constructor(element) {
+    this.element = element;
+  }
+
+  getBoundingClientRect() {
+    return addPadding(this.element.getBoundingClientRect(), 8);
+  }
+
+  get clientWidth() {
+    return this.getBoundingClientRect().width;
+  }
+
+  get clientHeight() {
+    return this.getBoundingClientRect().height;
+  }
+}
+
+const GapOverlay = ({ gap, isVisible, slide }) => {
+  const overlayLeft = {
+    top: `${gap.top}px`,
+    left: '0',
+    width: `${gap.left}px`,
+    height: `${gap.height}px`
+  };
+  const overlayRight = {
+    top: `${gap.top}px`,
+    left: `${gap.right}px`,
+    width: `calc(100% - ${gap.right}px)`,
+    height: `${gap.height}px`
+  };
+  const overlayTop = {
+    top: '0',
+    left: '0',
+    width: '100%',
+    height: `${gap.top}px`
+  };
+  const overlayBottom = {
+    top: `${gap.bottom}px`,
+    left: '0',
+    width: '100%',
+    height: `calc(100% - ${gap.bottom}px)`
+  };
+
+  return (
+    <React.Fragment>
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayLeft}
+      />
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayTop}
+      />
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayRight}
+      />
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayBottom}
+      />
+    </React.Fragment>
+  );
+};
+
+GapOverlay.propTypes = {
+  gap: PropTypes.shape({
```

Comment:
PropTypes.shape({...}).isRequired
Also every property should be required

Reply:
ok